lua/lspinstall/servers/terraform.lua

```diff
@@ -0,0 +1,45 @@
+local config = require'lspconfig'.terraformls.document_config
+require'lspconfig/configs'.terraformls = nil -- important, immediately unset the loaded config again
+config.default_config.cmd[1] = "./terraform-ls"
+
+return vim.tbl_extend('error', config, {
+  install_script = [[
+  os=$(uname -s | tr "[:upper:]" "[:lower:]")
+  arch=$(uname -m | tr "[:upper:]" "[:lower:]")
+
+  case $os in
+  linux)
+    platform="linux"
+  ;;
+  darwin)
+    platform="mac"
+  ;;
+  esac
+
+  case $arch in 
+  i386)
+    architecture=386
+  ;;
+  i586)
+    architecture=386
+  ;;
+  i686)
+    architecture=386
+  ;;
+  x86_64)
+    architecture=amd64
+  ;;
+  arm64)
+    architecture=arm64
+  ;;
+  aarch64)
+    architecture=arm64
+  ;;
+  esac
+  export PATTERN="browser_download_url.*$platform.*$architecture.*"
+  DOWNLOAD_URL=$(curl -s -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/hashicorp/terraform-ls/releases/latest | awk 'BEGIN { FS = "\"" }; $0 ~ ENVIRON["PATTERN"] { print $4 }')
+  curl -L -o terraform.zip "$DOWNLOAD_URL"
+  unzip terraform.zip
```

Comment:
Before unzipping do `rm terraform-ls | true`, this makes sure that the script can be run multiple times without unzip asking for confirmation to overwrite files.

Reply:
Tested this and it can result in it complaining on the first install so went with `rm -f` instead as that silently ignores files that don't exist and always returns 0.